1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,34 @@
-function setAlarm() {}
+let setTimeRemain = (inputValue) => {
+  let timeRemaining = document.getElementById("timeRemaining");
+  var hours = Math.floor(inputValue / 60);
+  var minutes = inputValue % 60;
+  timeRemaining.innerText = `Time Remaining: ${
+    (hours < 10 ? "0" + hours : hours) +
+    ":" +
+    (minutes < 10 ? "0" + minutes : minutes === 0 ? "00" : minutes)
+  }`;
+};
+
+let setBackgroundColor = () => {
+  var randomColor = Math.floor(Math.random() * 16777215).toString(16);
+  document.body.style.backgroundColor = "#" + randomColor;
+};
+
+function setAlarm() {
+  let inputValue = document.getElementById("alarmSet").value;
+  setTimeRemain(inputValue);
+  let timer_interval = setInterval(() => {
+    inputValue = inputValue - 1;
+    setTimeRemain(inputValue);
+    setBackgroundColor();
+    if (inputValue === 0) {
+      // inputValue = "";
+      clearInterval(timer_interval);
+      playAlarm();
+      document.body.style.backgroundColor = "#1c1c1c";
+    }
+  }, 1000);
+}
 
 // DO NOT EDIT BELOW HERE
 
```

Comment:
Hi Mahri! Just a few comments on this one, I would suggest making things are modular as possible and breaking into smaller functions. I also had a play round with trying to break things and I found that:
 - I could enter another time while the alarm was running and it would flick back and forth between both
 - I could input a negative number and it would try and count down with that.

You could try having a think about how you might deal with these cases!

Reply:
> Hi Mahri! Just a few comments on this one, I would suggest making things are modular as possible and breaking into smaller functions. I also had a play round with trying to break things and I found that:
> 
> * I could enter another time while the alarm was running and it would flick back and forth between both
> * I could input a negative number and it would try and count down with that.
> 
> You could try having a think about how you might deal with these cases!

@bec-gray oh yeah I think I need if else statement in here